MREUnityRuntime/MREUnityRuntimeLib/Core/Actor.cs

```diff
@@ -522,6 +533,56 @@ private RigidBody AddRigidBody()
             return RigidBody;
         }
 
+        private Collider SetCollider(ColliderPatch colliderPatch)
+        {
+            if (colliderPatch == null || colliderPatch.ColliderGeometry == null)
+            {
+                return null;
+            }
+
+            var colliderGeometry = colliderPatch.ColliderGeometry;
+            var colliderType = colliderGeometry.ColliderType;
+
+            if (_collider != null)
+            {
+                if (Collider.ColliderType == colliderType)
+                {
+                    // We have a collider already of the same type as the desired new geometry.
+                    // Update its values instead of removing and adding a new one.
+                    colliderGeometry.Patch(_collider);
+                }
+                else
+                {
+                    Destroy(_collider);
+                    _collider = null;
+                    Collider = null;
+                }
+            }
+
+            UnityCollider unityCollider = null;
+
+            switch (colliderType)
+            {
+                case ColliderType.Box:
+                    var boxCollider = gameObject.AddComponent<BoxCollider>();
```

Comment:
>var boxCollider = gameObject.AddComponent<BoxCollider>(); [](start = 20, length = 57)

If _collider wasn't destroyed above, would calling SetCollider again result in multiple Collider components on the actor?

Reply:
Yes it would.  Colliders are just components on the game object, and you can technically have as many colliders that you want on the game object.  They may not work as expected, but it is allowed by Unity.

Reply:
I see what you mean.  Good catch.  Updated to return the collider above in the case that we just patch it since it is already the correct type.